opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/aggregation/AggregationQueryBuilder.java

```diff
@@ -112,10 +112,26 @@ public Map<String, OpenSearchDataType> buildTypeMapping(
       List<NamedAggregator> namedAggregatorList,
       List<NamedExpression> groupByList) {
     ImmutableMap.Builder<String, OpenSearchDataType> builder = new ImmutableMap.Builder<>();
-    namedAggregatorList.forEach(agg -> builder.put(agg.getName(),
-        OpenSearchDataType.of(agg.type())));
-    groupByList.forEach(group -> builder.put(group.getNameOrAlias(),
-        OpenSearchDataType.of(group.type())));
+    namedAggregatorList.forEach(agg -> {
+      if (agg.type() == ExprCoreType.TIMESTAMP
+          || agg.type() == ExprCoreType.DATETIME
+          || agg.type() == ExprCoreType.DATE
+          || agg.type() == ExprCoreType.TIME) {
+        builder.put(agg.getName(), OpenSearchDateType.of(agg.type()));
+      } else {
+        builder.put(agg.getName(), OpenSearchDataType.of(agg.type()));
+      }
+    });
+    groupByList.forEach(group -> {
+      if (group.type() == ExprCoreType.TIMESTAMP
```

Comment:
Switch statement be nicer?

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/5f34b391f988e4f2e179a68bce688abc1dc15914